x-pack/legacy/plugins/uptime/public/components/functional/empty_state/data_or_index_missing.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiFlexGroup,
+  EuiEmptyPrompt,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiPanel,
+  EuiTitle,
+  EuiButton,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React, { useContext } from 'react';
+import { Link } from 'react-router-dom';
+import { UptimeSettingsContext } from '../../../contexts';
+import { DynamicSettings } from '../../../../common/runtime_types';
+import { SETTINGS_ROUTE } from '../../../../common/constants';
+
+interface DataMissingProps {
+  headingMessage: JSX.Element;
+  settings?: DynamicSettings;
+}
+
+const SETTINGS_LINK_TEXT = i18n.translate('xpack.uptime.empty_state.settingsLink', {
+  defaultMessage: 'Update the index pattern ',
+});
+
+export const DataOrIndexMissing = ({ headingMessage, settings }: DataMissingProps) => {
+  const { basePath } = useContext(UptimeSettingsContext);
+  return (
+    <EuiFlexGroup justifyContent="center" data-test-subj="data-missing">
+      <EuiFlexItem grow={false} style={{ flexBasis: 700 }}>
+        <EuiSpacer size="m" />
+        <EuiPanel>
+          <EuiEmptyPrompt
+            iconType="uptimeApp"
+            title={
+              <EuiTitle size="l">
+                <h3>{headingMessage}</h3>
+              </EuiTitle>
+            }
+            body={
+              <>
+                <p>
+                  <FormattedMessage
```

Comment:
What do we think about these `FormattedMessage` components? We've talked about exporting calls to `i18n.translate`, but these don't seem to clutter up the component nearly as much in terms of readability IMO. That said, I'd be fine with creating named FC exports from the same translation files.

CC @andrewvc

Reply:
yes i think we can keep formatted messages inline to jsx.

Reply:
I'm fine with these as-is, but don't have a strong opinion